helper.go

```diff
@@ -556,7 +583,8 @@ func parseInt(bytes []byte) (int, error) {
 // * Long form. Two to 127 octets. Bit 8 of first octet has value "1" and bits
 //   7-1 give the number of additional length octets. Second and following
 //   octets give the length, base 256, most significant digit first.
-func parseLength(bytes []byte) (length int, cursor int) {
+func parseLength(bytes []byte) (length int, cursor int, err error) {
+	var ErrInvalidPacketLength = errors.New("invalid packet length")
```

Comment:
Should we make this a global var so it doesn't get generated for every function call?